src/main/kotlin/org/rust/lang/core/psi/ext/RsLifetimeParameter.kt

```diff
@@ -0,0 +1,18 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.psi.ext
+
+import org.rust.lang.core.psi.RsLifetime
+import org.rust.lang.core.psi.RsLifetimeParameter
+
+val RsLifetimeParameter.bounds: List<RsLifetime> get() {
+    val owner = parent?.parent as? RsGenericDeclaration
+    val whereBounds =
+        owner?.whereClause?.wherePredList.orEmpty()
+            .filter { it.lifetime?.quoteIdentifier?.text == this.quoteIdentifier.text }
```

Comment:
Hm, I think we should use `resolve` here
Something like this: `.filter { it.lifetime?.reference?.resolve() == this }`